modules/charon-core/src/main/java/org/wso2/charon3/core/encoder/JSONDecoder.java

```diff
@@ -917,9 +921,15 @@ public BulkRequestData decodeBulkData(String bulkResourceString) throws BadReque
 
                 if (requestMethod.equals(SCIMConstants.OperationalConstants.POST)) {
 
-                    if (!member.optString(SCIMConstants.OperationalConstants.BULK_ID).equals("") &&
-                            member.optString(SCIMConstants.OperationalConstants.BULK_ID) != null) {
+                    String bulkId = member.optString(SCIMConstants.OperationalConstants.BULK_ID);
 
+                    if (StringUtils.isNotEmpty(bulkId)) {
+                        if (encounteredBulkIds.contains(bulkId)) {
+                            String error = "Duplicate bulkId found: " + bulkId;
+                            logger.error(error);
```

Comment:
No need to log the error here, The place which catch and handle the exception will log the error. If we log and throw here, there will be two duplicated error logs

Reply:
Removed